lib/pbench/server/indexing_tarballs.py

```diff
@@ -524,67 +565,49 @@ def sighup_handler(*args):
                                 raise
                             except Exception:
                                 pass
-                        # Distinguish failure cases, so we can retry the indexing
-                        # easily if possible.  Different `linkerrdest` directories for
-                        # different failures; the rest are going to end up in
-                        # `linkerrdest` for later retry.
-                        controller_path = linksrc_dir.parent
 
+                        # Distinguish failure cases, so we can retry the indexing
+                        # easily if possible.
+                        #
+                        # Only if the indexing was successful do we request the
+                        # next operation (tool indexing). Otherwise we record
+                        # the error in the `server.errors.index` metadata and
+                        # leave the dataset in INDEXING state. (FIX-ME: should
+                        # we expose an "INDEX_FAILED" primary state which would
+                        # "trap" the failed dataset but enable a transition to
+                        # INDEXING to re-try?)
                         if tb_res is error_code["OK"]:
```

Comment:
Should this be `if tb_res.success`?

Reply:
Not my monkey, but I'll put a leash on it. 👍🏻